### PR TITLE
Dont pass the table to the header object, but generate it in ::new()

### LINKED ISSUE #127: store: EntryHeader should build a standard-header
I guess the `EntryHeader::new()` function should not get a `table` passed but build its own one and set some default entities within.

What do you think?

This is an interface-change!

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Closes #127 

I will not merge this yet, as we have to build the default-header. I will implement it as far as the docs say by now, but maybe we want more...

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
It might be even better to implement `Default` and not providing a `::new()` method,... not sure.

@TheNeikos ?

**Comment by TheNeikos** [comment]
Implementing `Default` here makes not a lot of sense in my opinion. I would just not make it public and instead let the entry create one (with the defaults)

**Comment by matthiasbeyer** [comment]
So you would do it as I did in my PR? :smile:

**Comment by TheNeikos** [comment]
Yes, except the 'hardcoded' part in the version

**Comment by matthiasbeyer** [comment]
Yes, I really should change that to the `version!()` of the crate, right? :-)

File: libimagstore/src/header.rs [hunk 1]
```diff
@@ -30,3 +36,20 @@ impl EntryHeader {
...
+
+fn build_default_header() -> BTreeMap<String, Value> {
+    let mut m = BTreeMap::new();
+
+    m.insert(String::from("imag"), {
+        let mut imag_map = BTreeMap::<String, Value>::new();
+
+        imag_map.insert(String::from("version"), Value::String(String::from("hardcoded")));
+        imag_map.insert(String::from("links"), Value::Array(vec![]));
+        imag_map.insert(String::from("content"), Value::Table(BTreeMap::new()));
```
Comment (TheNeikos) [inline libimagstore/src/header.rs]: Don't put it in per default! It should only be present if the content is not in the file!

File: libimagstore/src/header.rs [hunk 2]
```diff
@@ -30,3 +36,20 @@ impl EntryHeader {
...
 
 }
+
+fn build_default_header() -> BTreeMap<String, Value> {
+    let mut m = BTreeMap::new();
+
+    m.insert(String::from("imag"), {
+        let mut imag_map = BTreeMap::<String, Value>::new();
+
+        imag_map.insert(String::from("version"), Value::String(String::from("hardcoded")));
```
Comment (TheNeikos) [inline libimagstore/src/header.rs]: hardcoded? Perhaps use version! or something similar?

**Comment by matthiasbeyer** [comment]
@TheNeikos review pls.

**Comment by TheNeikos** [comment]
ACK

**Comment by matthiasbeyer** [comment]
Waiting on CI.

**Pull request merged**